pkg/startup/startup.go

```diff
@@ -555,7 +555,11 @@ func configureIPsAndSubnets(node *api.Node) (bool, error) {
 
 	if ipv4Env == "none" && (ipv6Env == "" || ipv6Env == "none") && node.Spec.BGP.IPv4Address == "" && node.Spec.BGP.IPv6Address == "" {
 		log.Warn("No IP Addresses configured, and autodetection is not enabled")
-		terminate()
+		// Unrecoverable if Calico is handling networking, terminate to restart.
+		if os.Getenv("CALICO_NETWORKING_BACKEND") != "none" {
+			terminate()
+		}
+		return false, nil
```

Comment:
I don't think you should return false here - strictly according to the method doc we should still run through the code below to determine if the IPs have changed.

Reply:
hmm, but that means if an IP address is empty (which is OK now in some cases) will be compared to previously set value. Is that necessary? What condition are you thinking this will be necessary to IP address comparison for?

Reply:
I don't really have anything in mind - other than trying to simplify the flow and keeping the behavior true to the function signature.